syntax = "proto3";
package pulse.cdm.bind;
option java_package = "com.kitware.pulse.cdm.bind";
option csharp_namespace = "pulse.cdm.bind";
option optimize_for = SPEED;

import "pulse/cdm/bind/Enums.proto";
import "pulse/cdm/bind/Properties.proto";
import "pulse/cdm/bind/Substance.proto";

message ECMOSettingsData
{
  eSwitch                               Connection                                = 1;/**<< @brief @ref eSwitchTable */
  ScalarVolumeData                      PrimingVolume                             = 2;/**<< @brief Includes the volume in the device and tubing. */
  ScalarVolumePerTimeData               TransfustionFlow                          = 3;/**<< @brief Total compliance of the entire mechanical ventilator circuit. */
  
  oneof Primer
  {
    string                              SubstanceCompound                         = 4;/**<< @brief Set volume to a specific substance compound. */
    SubstanceConcentrationListData      SubstanceConcentrationList                = 5;/**<< @brief Set the volume with several substance concentrations. */
  }
}

message ECMOData
{
  ECMOSettingsData                      Settings                                  = 1;
}
